course6/toyapp-vivike/displaylib/surface.cc

```diff
@@ -24,57 +26,62 @@ void Surface::Clear(Color backgroundColor)
 	for(int y = 0; y < mSurface->h; y++)
 		for(int x = 0; x < mSurface->w; x++)
 		{
-			Vector2D pixel;
-			pixel.x = x;
-			pixel.y = y;
+			Pixel pixel{x, y};
 			SetPixel(pixel, backgroundColor);
 		}
 }
 
-void Surface::SetPixel(Vector2D pixel, Color currentColor)
+void Surface::SetPixel(Pixel& pixel, Color currentColor)
 {
 	assert(mSurface);
 
 	std::uint32_t pixelValue = (std::uint32_t(currentColor.R)<<16) + (std::uint32_t(currentColor.G)<<8) + currentColor.B;
 
-	std::uint32_t* pixelAddr = static_cast<std::uint32_t*>(mSurface->pixels) + (pixel.y*mSurface->pitch/4 + pixel.x);
+	std::uint32_t* pixelAddr = static_cast<std::uint32_t*>(mSurface->pixels) + (pixel.vec[1]*mSurface->pitch/4 + pixel.vec[0]);
 	*pixelAddr = pixelValue;
 }
 
-void Surface::DrawLine(Vector2D p1, Vector2D p2, Color color)
+void Surface::DrawLine(Pixel& p1, Pixel& p2, Color color)
 {
 	// Bresenham's line algorithm
-	const bool steep = (fabs(p2.y - p1.y) > fabs(p2.x - p1.x));
+	int v1x = p1.vec[0];
```

Comment:
This is awkward API. It would be better to implement ```operator[]``` in your vector, so that uses can use it like that:
```
int v1x = p1[0];
```